content/300-guides/300-upgrade-guides/300-upgrading-to-use-preview-features/050-enabling-named-constraints.mdx

```diff
@@ -6,68 +6,63 @@ metaDescription: 'How to use the Preview feature `namedConstraints`.'
 
-<!-- keep names -->
-<!--
-- set flag and introspect your target env
--->
+The names of compound unique constraints (`@@unique`) were passed directly into the client after introspecting a database. This led to problems if these contained characters that resulted in invalid Typescript types.
 
-<!-- switch to default names / keep schema clean -->
-<!--
-- use clean schema, call migrate dev
-- apply to other envs
--->
+The Prisma schema did not contain syntax to describe the names of primary keys and foreign keys. This could lead to situations where introspecting an existing database and then replicating it locally based on the introspected schema yielded databases that differed in their constraint names. This results in failing migrations if later on these
+constraints are manipulated since they are addressed by name.
 
-## Upgrade paths for Prisma Migrate users
+If you started a development environment based on an initial introspection of a database you should [check whether their constraint names match](#checking-whether-your-environments-use-identical-names)! If the names are out of sync you can now [align them by following these steps](#aligning-your-environments-if-their-constraint-or-index-names-differ).
 
-Use the following guide if your project uses Prisma Migrate. Choose the upgrade path that matches your requirements:
+## Upgrade path for introspection only users
 
-- [I want to maintain my existing constraint and index names](#upgrade-path-1-i-want-to-maintain-my-existing-constraint-and-index-names)
-- [I want to use Prisma's default constraint and index names](#upgrade-path-2-i-want-to-use-prismas-default-constraint-and-index-names)
+Use the following guide if you use introspection only.
+
+1. Enable the `namedConstraints` Preview:
+
+   ```prisma highlight=3;normal
+   generator client {
+     provider        = "prisma-client-js"
+     previewFeatures = ["namedConstraints"]
+   }
+   ```
+
+1. Introspect to populate your Prisma schema with any constraint names that do not follow Prisma's naming convention:
+
+   ```terminal
+   npx prisma introspect
+   ```
+
+   For example, in the following model, the `@id` primary key constraint does not follow Prisma's convention (the conventional name would be `User_pk`):
+
+   ```prisma
+   model User {
+     id    Int    @id(map: "user_id_primary_key") @default(autoincrement())
+     name  String @unique
+     posts Post[]
+   }
+   ```
+
+## Upgrade paths for Prisma migrate users
+
+When enabling the namedConstraints preview flag, you are changing the behaviour of Prisma migrate. The default naming convention for constraint and index names will change and you primary and foreign key names will now be part of the schema for databases that support them. Therefore the meaning of your existing schema to Prisma will change. Before you continue to evolve your schema and your database, you should decided which names for constraints and indexes you want to use on your project from now on.
```

Comment:
Split this section into paragraphs

When enabling the `namedConstraints` preview flag, you are changing the behavior of Prisma migrate. 

The default naming convention for constraint and index names will change and your primary and foreign key names will now be part of the schema for databases that support them. Therefore the meaning of your existing schema to Prisma will change. 

Before you continue to evolve your schema and your database, you should decided which names for constraints and indexes you want to use on your project from now on.

Reply:
Done